stanford-earth/SE3_BLT
# EARTH-256: Add module and profile behat paths

# DO NOT MERGE

# Summary
- Changes to the `blt/project.yml` file should help BLT generate a local.yml file that includes additional paths, like `/modules` and `/profiles`, to look for behat tests.
- It also allows us to output the results as an html page, in addition to the `pretty` progress report.  I like this because it saves the results past your terminal sessions and running Behat via Ansible gives us less control on the stout.
- We discussed replacing `tests:behat` or creating a new, `sws:behat`. I like creating an `sws:` prefix for BLT commands we override.  That way, if we don't have to wonder, did we end up overriding this?  It's incredibly clear.

# Needed By (8/14)
- It would be great to have this merged prior to Aug. 14th, when I am scheduled to punch a minor release of Caravan.

# Steps to Test
- Setup requires a bit of work.  Checkout this branch.
- Place a feature file, somewhere like: `docroot/modules/stanford/stanford_news/tests/stanford_news.feature` from https://raw.githubusercontent.com/SU-SWS/linky_clicky/5.x/sites/uat/features/high-value/stanford_news.feature.
- Run `blt sws:behat` and it should find the stanford_news.feature file.
- The test will fail for several reasons, number 1 being that linky_clicky is not being pulled in as a package because we are not ready to change the default branch to 6.x, where we're building out the D8 package of SWS step definitions.
- You should be able to see output from the tests as they run and afterwards find the full results at http://[local_url]/reports/behat/index.html.

# Affected versions or modules
- This affects the next release of Caravan, scheduled for Aug. 14th.

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/EARTH-256
